server/src/modules/clubs/application/use-cases/get-all-clubs/get-all-clubs-controller.ts

```diff
@@ -7,7 +7,7 @@ export class GetAllClubsController extends ControllerWithoutDTO<GetAllClubsUseCa
     super(useCase)
   }
 
-  async execute(_req: express.Request, res: express.Response): Promise<express.Response> {
+  async execute<Res extends express.Response>(_req: express.Request, res: Res): Promise<Res> {
```

Comment:
Previously, this would return a `Promise<express.Response>`, even if we passed in a more specific object. Now, the return type is the same as the type of `res`, so we can pass in more specific things (like the `httpmocks` mock response object) and it'll forward that type when it returns